homeassistant/components/thesilentwave/binary_sensor.py

```diff
@@ -0,0 +1,63 @@
+"""Support for TheSilentWave sensors."""
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .coordinator import TheSilentWaveCoordinator
+from .entity import TheSilentWaveEntity
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor from a config entry."""
+    coordinator = entry.runtime_data
+    async_add_entities([TheSilentWaveBinarySensor(coordinator, entry.entry_id)], True)
+
+
+class TheSilentWaveBinarySensor(TheSilentWaveEntity, BinarySensorEntity):
+    """Representation of a TheSilentWave binary sensor."""
+
+    _attr_translation_key = "status"
+
+    def __init__(self, coordinator: TheSilentWaveCoordinator, entry_id: str) -> None:
+        """Initialize the binary sensor."""
+        super().__init__(coordinator, entry_id)
+        self._attr_name = None  # Set to None to use device name as this is the main sensor
+        self._unsubscribe_callback = None
+        # Set a more specific unique_id for this sensor entity
+        self._attr_unique_id = f"{DOMAIN}_{entry_id}_status"
```

Comment:
unique ids are unique per domain per platform, so no need to incorporate `DOMAIN` in there